numcodecs/tests/test_entrypoints.py

```diff
@@ -19,6 +19,27 @@ def set_path():
     numcodecs.registry.codec_registry.pop("test")
 
 
-def test_entrypoint_codec(set_path):
+@pytest.fixture()
+def set_path_fixture():
+    yield from set_path()
+
+
+@pytest.mark.usefixtures("set_path_fixture")
+def test_entrypoint_codec():
     cls = numcodecs.registry.get_codec({"id": "test"})
     assert cls.codec_id == "test"
+
+
+def test_entrypoint_codec_with_importlib_metadata():
+    # importlib_metadata patches importlib.metadata, which can lead to breaking changes
+    # to the APIs of EntryPoint objects used when registering entrypoints. Attempt to
+    # isolate those changes to just this test.
+    with mock.patch.dict(sys.modules):
```

Comment:
In hindsight, this may not really be enough to avoid this import polluting the rest of the test suite. Even if `importlib_metadata` is removed from `sys.modules`, the monkey-patching is already done. We might need to launch this test in a subprocess to really avoid this problem...

Reply:
OK I have a version of this test locally that's rewritten to use `multiprocessing.Process` and will fully isolate the import. I can push that version if maintainers are OK with a subprocess in the test.

Reply:
All right, since there were some unrelated test failures I pushed out the multiprocessing version anyway so that you can take a look. I can revert if it's preferred to avoid that.